fast/stages/0-org-setup/datasets/hardened/organization/custom-constraints/custom.firewallRestrictCacheSearchDatabasesPolicyRule.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+custom.firewallRestrictCacheSearchDatabasesPolicyRule:
+  action_type: DENY
+  condition: |-
+    resource.rules.exists(rule,
+      rule.priority < 2147483644 &&
+      rule.direction == 'INGRESS' &&
+      rule.match.srcIpRanges.exists(range, range == '0.0.0.0/0') &&
+      rule.match.layer4Configs.exists(l4config,
+        (l4config.ipProtocol == 'tcp' || l4config.ipProtocol == 'udp') &&
+        l4config.ports.exists(port,
+          port == '9200' || port == '9300' || port == '11211'
+          || port == '11214' || port == '11215' || port == '6379'
+        )
```

Comment:
Those rules may need to check using `containsFirewallPort` instead of equality, to cover also the cases where ports are provided as a range.